src/Microsoft.OData.Client/ALinq/QueryableResourceExpression.cs

```diff
@@ -371,8 +372,24 @@ internal void AddFilter(IEnumerable<Expression> predicateConjuncts)
             this.keyPredicateConjuncts.Clear();
         }
 
-        internal void AddApply(Expression aggregateExpr, OData.UriParser.Aggregation.AggregationMethod aggregationMethod)
+        /// <summary>
+        /// Adds an aggregation expression to the $apply expression of this resource expression.
+        /// If the resource expression already contains a $filter expression, 
+        /// the predicate conjuncts of the $filter expression are moved
+        /// to the predicate conjuncts of the $apply expression and 
+        /// the $filter expression removed.
+        /// The predicate conjuncts in the $apply expression are used to generate 

```

Comment:
@habbes We apply a filter transformation to the `$apply` query option. Take the following expression for instance:
```csharp
dataServiceContext.Sales.Where(d0 => d0.CurrencyCode == "USD").GroupBy(
    d1 => d1.Product.Color,
    (d2, d3) => new
    {
        ProductAvgTaxRate = d3.Average(d4 => d4.Product.TaxRate)
    });
```
This will be translated into the following OData URL:
```
/Sales?$apply=filter(CurrencyCode eq 'USD')/groupby((Product/Color),aggregate(Product/TaxRate with average as ProductAvgTaxRate))
```
Having both `$filter` and `$apply` query options is NOT allowed